### PR TITLE
fix(hygiene): recognize CJK sentence boundaries in dump heuristics

### PR DESCRIPTION
**Pull request by ddddddddwp**
## Summary

Addresses #806 (Lane 2).

The hygiene audit and strict write classifier both used `content.count(". ")` to estimate sentence structure. That misclassified long CJK prose and ASCII prose whose sentences ended at line boundaries as `likely_dump`.

This PR:

- adds one shared sentence-boundary counter;
- recognizes CJK `。！？` terminators;
- recognizes ASCII `.`, `!`, and `?` when followed by whitespace or end-of-content, including line-ending punctuation;
- keeps the existing hygiene/write line-count thresholds and dump reason contracts unchanged;
- preserves rejection of long low-structure multiline dumps;
- adds regression tests for CJK punctuation, ASCII line-ending punctuation, and the existing dump control;
- updates `CHANGELOG.md`.

Lane 3 acknowledgement/log retention policy is intentionally out of scope.

## Verification

- `MNEMOSYNE_NO_EMBEDDINGS=1 .venv/bin/python -m pytest tests/test_filters.py tests/test_hygiene.py -n auto -q` -> 260 passed
- `ruff check mnemosyne/core/filters.py mnemosyne/core/hygiene.py tests/test_filters.py tests/test_hygiene.py` -> passed
- `git diff --check` -> passed

A full `tests -n auto` run reached 3299 passed, 32 failures, 6 skipped, and 7 subtests passed. The failures are pre-existing parallel/environment-sensitive failures in stats, config-profile, and benchmark-helper tests; they do not touch the changed files. The affected suites pass independently.

## Summary

- Improves dump detection for CJK (`。！？`) and ASCII (`. ! ?`) sentence boundaries.
- Preserves existing hygiene thresholds, strict-write thresholds, and dump-reason contracts.
- Keeps long, low-structure multiline dumps rejected.
- Adds shared sentence-boundary logic and regression coverage.
- Updates `CHANGELOG.md`.

## Architectural Impact

- No changes to working, episodic, or BEAM memory tiers.
- No changes to retrieval, consolidation, veracity, synchronization, or benchmark methodology.
- No changes to Hermes, MCP, or CLI integration surfaces.
- No privacy or local-first impact. The change remains local to memory-write heuristics.
- Shared boundary detection removes duplicated logic and improves long-term maintainability. This is the right call.

### CODE REVIEW

File: tests/test_filters.py [hunk 1]
```diff
@@ -323,6 +323,16 @@ def test_rejects_large_dump(self):
...
 
+    @pytest.mark.parametrize("terminator", ["。", "！", "？", "."])
+    def test_allows_multiline_sentences_with_cjk_or_line_end_punctuation(self, terminator):
+        content = "\n".join(
+            [f"This is a complete sentence with useful content{terminator}"] * 60
+        )
+
+        decision = classify_memory_write(content)
+
+        assert decision.action == "allow"
```
[the remark this answers is not in the document]
Comment (ddddddddwp) [reply tests/test_filters.py]: Fixed in df494ff. The regression matrix now covers ASCII ! and ?, and both filters and hygiene tests include dotted low-structure dump fixtures with assertions for the expected reject reason and likely_dump signal. Focused tests: 266 passed; Ruff passed.

**Comment by ddddddddwp** [comment]
Rebased the branch onto the latest `origin/main` (`8e6c010`) and force-pushed the updated head (`355f9ca`). The two original commits were preserved without patch changes. Local verification: `266 passed` for `tests/test_filters.py` and `tests/test_hygiene.py` with `-n auto`; Ruff 0.15.22 passed for the affected source and test files. The existing `[Unreleased]` changelog entry remains included.

**Comment by dplush** [comment]
Thanks for addressing the sentence-boundary regression coverage.

I reviewed the current head. The shared helper correctly handles CJK and ASCII sentence boundaries, while dotted paths, versions, and identifiers remain classified as low-structure dumps. The focused tests and CI are green, and I found no blocking issues.

Please rebase onto the latest `main` and rerun CI. After that, this is good to merge.

**Comment by ddddddddwp** [comment]
Thanks for the review. Rebased onto the latest `main` (`3bb08dd`) and force-pushed the updated head `13f2f08`; the two commits were preserved without patch changes.

Local verification on the rebased head: `MNEMOSYNE_NO_EMBEDDINGS=1 .venv/bin/python -m pytest tests/test_filters.py tests/test_hygiene.py -n auto -q` -> 266 passed, and Ruff 0.15.22 is clean for the affected source and test files. The `CHANGELOG.md` entry for `#806` remains included. Cloud CI will re-run on this head.

**Comment by dplush** [comment]
Reviewed the current head `13f2f08`.

No blocking issues found. The shared sentence-boundary helper is correctly wired into both heuristics, and the added CJK, ASCII, and dotted-dump coverage is appropriate.

Local verification passed:

- 266 focused tests passed
- `compileall` passed
- `git diff --check` passed
- GitHub CI is green across Python 3.10 to 3.13, lint, build, docs, and CLA

One minor edge case remains: structured lines ending in `?` or `!`, for example URLs such as `https://example.test/api?`, are counted as sentence boundaries and could bypass `likely_dump` detection. This is not blocking for this PR, but a regression test could be useful if such inputs are in scope.

From a code perspective, this is ready to merge.
